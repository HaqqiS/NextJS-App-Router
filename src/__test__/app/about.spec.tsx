import AboutPage from "@/app/about/page";
import AboutLayout from "@/app/about/layout";
import { render, screen } from "@testing-library/react";
import { describe } from "node:test";

describe("AboutPage", () => {
    it("should render the about page", () => {
        const page = render(
            <AboutLayout>
                <AboutPage />
            </AboutLayout>
        );

        expect(page).toMatchSnapshot();
    });
});
