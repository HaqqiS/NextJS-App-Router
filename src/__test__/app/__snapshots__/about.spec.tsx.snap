// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the about page 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <nav
        class="bg-slate-600 fixed right-0 top-14 w-60 h-screen z-10 text-white"
      >
        <ul
          class="flex flex-col gap-3 px-5 pt-4"
        >
          <li>
            Home
          </li>
          <li>
            About
          </li>
          <li>
            Profile
          </li>
        </ul>
      </nav>
      <div>
        <div>
          <h1>
            AboutPage
          </h1>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <nav
      class="bg-slate-600 fixed right-0 top-14 w-60 h-screen z-10 text-white"
    >
      <ul
        class="flex flex-col gap-3 px-5 pt-4"
      >
        <li>
          Home
        </li>
        <li>
          About
        </li>
        <li>
          Profile
        </li>
      </ul>
    </nav>
    <div>
      <div>
        <h1>
          AboutPage
        </h1>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
